pkg/controller/node/cidr_set.go

```diff
@@ -35,13 +35,28 @@ type cidrSet struct {
 	subNetMaskSize  int
 }
 
+const (
+	// The subnet mask size cannot be greater than 16 more than the cluster mask size
+	// TODO: https://github.com/kubernetes/kubernetes/issues/44918
+	// clusterSubnetMaxDiff limited to 16 due to the uncompressed bitmap
+	clusterSubnetMaxDiff = 16
+	// maximum 64 bits of prefix
+	maxPrefixLength = 64
+)
+
 func newCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *cidrSet {
 	clusterMask := clusterCIDR.Mask
 	clusterMaskSize, _ := clusterMask.Size()
-	maxCIDRs := 1 << uint32(subNetMaskSize-clusterMaskSize)
+
+	var maxCIDRs int
+	if ((clusterCIDR.IP.To4() == nil) && (subNetMaskSize-clusterMaskSize > clusterSubnetMaxDiff)) || (subNetMaskSize > maxPrefixLength) {
+		maxCIDRs = 0
```

Comment:
ok, I will add this. This will cause a change in the calling function, so I will add this change to PR#45792
which is dependent on this commit.